implementations/micrometer-registry-signalfx/src/main/java/io/micrometer/signalfx/SignalfxDistributionSummary.java

```diff
@@ -16,88 +16,69 @@
- * cumulative.
+ * A StepDistributionSummary which provides support for multiple flavours of Histograms to
+ * be recorded based on {@link SignalFxConfig#publishCumulativeHistogram()} and
+ * {@link SignalFxConfig#publishDeltaHistogram()}.
  *
  * @author Bogdan Drutu
  * @author Mateusz Rzeszutek
+ * @author Lenin Jaganathan
  */
-final class SignalfxDistributionSummary extends AbstractDistributionSummary {
-
-    private final LongAdder count = new LongAdder();
-
-    private final DoubleAdder total = new DoubleAdder();
-
-    private final StepTuple2<Long, Double> countTotal;
-
-    private final TimeWindowMax max;
+final class SignalfxDistributionSummary extends StepDistributionSummary {
 
     @Nullable
-    private final DeltaHistogramCounts deltaHistogramCounts;
+    private final StepBucketHistogram stepBucketHistogram;
 
     SignalfxDistributionSummary(Id id, Clock clock, DistributionStatisticConfig distributionStatisticConfig,
             double scale, long stepMillis, boolean isDelta) {
-        super(id, clock, CumulativeHistogramConfigUtil.updateConfig(distributionStatisticConfig), scale, false);
-        this.countTotal = new StepTuple2<>(clock, stepMillis, 0L, 0.0, count::sumThenReset, total::sumThenReset);
-        max = new TimeWindowMax(clock, distributionStatisticConfig);
-        if (distributionStatisticConfig.isPublishingHistogram() && isDelta) {
-            deltaHistogramCounts = new DeltaHistogramCounts();
+        super(id, clock, distributionStatisticConfig, scale, stepMillis, defaultHistogram(clock,
+                CumulativeHistogramConfigUtil.updateConfig(distributionStatisticConfig, isDelta), false));
+
+        double[] slo = distributionStatisticConfig.getServiceLevelObjectiveBoundaries();
+        if (slo != null && slo.length > 0 && isDelta) {
+            stepBucketHistogram = new StepBucketHistogram(clock, stepMillis,
+                    DistributionStatisticConfig.builder()
+                        .serviceLevelObjectives(CumulativeHistogramConfigUtil.addPositiveInfBucket(slo))
+                        .build()
+                        .merge(distributionStatisticConfig),
+                    false, true);
         }
         else {
-            deltaHistogramCounts = null;
+            stepBucketHistogram = null;
         }
     }
 
     @Override
     protected void recordNonNegative(double amount) {
-        count.increment();
-        total.add(amount);
-        max.record(amount);
+        if (stepBucketHistogram != null) {
+            stepBucketHistogram.recordDouble(amount);
+        }
+        super.recordNonNegative(amount);
     }
 
     @Override
     public long count() {
-        return countTotal.poll1();
-    }
-
-    @Override
-    public double totalAmount() {
-        return countTotal.poll2();
-    }
-
-    @Override
-    public double max() {
-        return max.poll();
+        if (stepBucketHistogram != null) {
+            stepBucketHistogram.poll();
```

Comment:
A comment about why this is here would be good for more easily understanding it in the future while looking at the code. Same in the Timer implementation.